### PR TITLE
feat(pie):  pie chart type and legend clickable

### PR DESCRIPTION
**Pull request by justinZan**
## Description

> add pie chart type 
> pie chart can support legend filter by config

## Screenshots

截屏2023-08-01 11 17 57

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation or example update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation and example

### CODE REVIEW

File: src/components/gauge-chart/gauge-chart.ts [hunk 1]
```diff
@@ -61,24 +63,10 @@ class GaugeChart extends LitElement {
...
-      aspectRatio: this.options?.aspectRatio ?? defaultGaugeChartOptions.aspectRatio,
-      layout: this.options?.layout ?? defaultGaugeChartOptions.layout,
-      plugins: {
-        legend: {
-          display: this.options?.plugins?.legend?.display ?? defaultGaugeChartOptions.plugins?.legend?.display,
-        },
-      },
-    };
-
+    this.chartOptions = merge({}, defaultGaugeChartOptions, globalStyleOptions, this.options);
```
Comment (bndynet) [inline src/components/gauge-chart/gauge-chart.ts:68]: The `defaultGaugeChartOptions` should be the higher priority than `globalStyleOptions`.

File: src/components/pie-chart/pie-chart.types.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { ChartDataset, ChartOptions } from 'chart.js/auto';
+
+export interface PieChartData extends ChartDataset<'pie', number[]> {
+  centerValue?: string | number;
```
Comment (bndynet) [inline src/components/pie-chart/pie-chart.types.ts:4]: Can we rename it to **centerLabel**?

File: src/components/pie-chart/pie-chart.types.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+}
+
+export interface PieChartOptions extends ChartOptions<'pie'> {
+  type?: string;
+  theme?: string | [];
+  isLegendClick?: boolean;
+  // eslint-disable-next-line @typescript-eslint/ban-types
+  legendClickCallback?: Function;
```
Comment (bndynet) [inline src/components/pie-chart/pie-chart.types.ts:12]: We should specify the exact arguments here.  For example:
```
onLegendClick: (legendItem: {label: string, value: number}) => void;
```

File: src/core/common/global-style.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  'sequential-colors-health-green': ['#78F5B8', '#31E88C', '#00CF64', '#00AB50', '#00853C', '#03612C', '#08421F'],
+  'sequential-colors-health-yellow': ['#FFD98C', '#FFC14F', '#FC9D03', '#D97F00', '#A85F00', '#7D4705', '#54330D'],
+  'sequential-colors-health-red': ['#FFD5CC', '#FFBBAD', '#FF9580', '#F7644A', '#D4371C', '#A12512', '#6E1D13'],
+  'diverging-colors-alerts': ['#FFCD59', '#F7A947', '#EE8335', '#E55920', '#CF3110', '#A82013', '#821016'],
+  'diverging-colors-green-to-red': ['#008672', '#4BA28C', '#8CBBAC', '#CCD2CE', '#E19D79', '#C3411D', '#990000'],
+};
+
+export interface GlobalStyleOptions extends ChartOptions {
+  fontColor?: string;
+  fontFamily?: string;
```
Comment (bndynet) [inline src/core/common/global-style.ts:20]: Those properties should be placed into **ChartOptions**. And then the **GlobalStyleOptions** is no need.

File: src/core/common/global-style.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+}
+
+const globalStyleOptions: GlobalStyleOptions = {
```
Comment (bndynet) [inline src/core/common/global-style.ts:23]: By above change, here can be:
```js
const globalChartOptions: ChartOptions = { ... }
```

File: src/components/pie-chart/pie-chart.types.ts (hunk 2 above)
Comment (justinZan) [reply src/components/pie-chart/pie-chart.types.ts:4]: I will rename in next commit

**Pull request merged**